stdlib/pathlib.pyi

```diff
@@ -101,7 +100,10 @@ class PurePosixPath(PurePath): ...
 class PureWindowsPath(PurePath): ...
 
 class Path(PurePath):
-    def __new__(cls, *args: StrPath, **kwargs: Any) -> Self: ...
+    if sys.version_info >= (3, 12):
+        def __new__(cls, *args: StrPath, **kwargs: Unused) -> Self: ...  # pyright: ignore[reportInconsistentConstructor]
+    else:
+        def __new__(cls, *args: StrPath, **kwargs: Unused) -> Self: ...
```

Comment:
(The return type seems wrong to me. It should be `WindowsPath` or `PosixPath`, depending on the platform. But this is out of scope of this PR.)